src/groth16.cpp

```diff
@@ -288,8 +288,30 @@ Prover<Engine>::prove(typename Engine::FrElement* wtns)
     typename Engine::FrElement s;
     typename Engine::FrElement rs;
 
-    E.fr.copy(r, E.fr.zero());
-    E.fr.copy(s, E.fr.zero());
+    //E.fr.copy(r, E.fr.zero());
+    //E.fr.copy(s, E.fr.zero());
+
+
+    FrRawElement fr_modulus = {0x43E1F593F0000001ull, 0x2833E84879B97091ull,
```

Comment:
Where did you get these constants from and can you somehow assert against a library-defined constant that this is the right modulus in the right endianness?

Reply:
@alinush The number was provided by @mstraka100, I believe it was from the Groth Paper (@mstraka100 can you confirm). I did make sure it is encoded in the right endianness:

1. The fact that we used to have last byte set to zero suggested little endian
2. I double confirmed it by testing Fr_rawCmp with 0x1000 vs 0x0100 vs 0x0001, etc.

Reply:
It was taken from the arkworks BN128 implementation - I will add a comment stating this and cross-check it with other codebases/documentation to be sure

Reply:
Added a comment